srml/balances/src/lib.rs

```diff
@@ -620,6 +620,6 @@ where
 	T::Balance: MaybeSerializeDebug
 {
 	fn is_dead_account(who: &T::AccountId) -> bool {
-		Self::total_balance(who).is_zero()
+		!<FreeBalance<T>>::exists(who) && !<ReservedBalance<T>>::exists(who)
```

Comment:
Under what conditions do you expect these not to be equivalent?

Reply:
When existential deposit >= 1 they will always be equivalent because if the balances become too low, the FreeBalance::remove() and ReservedBalance::remove() will be called to remove the account and calling total_balance(who) will return zero.

The condition where these two are not equivalent is when existential deposit is equal to zero, with the expectation that an account will never be considered closed/dead after it has received a transfer, even if its total_balance goes to zero at some later time.